worker/metrics/collect/context.go

```diff
@@ -36,7 +36,8 @@ func (ctx *hookContext) HookVars(paths context.Paths) ([]string, error) {
 		"CHARM_DIR=" + paths.GetCharmDir(), // legacy
 		"JUJU_CHARM_DIR=" + paths.GetCharmDir(),
 		"JUJU_CONTEXT_ID=" + ctx.id,
-		"JUJU_AGENT_SOCKET=" + paths.GetJujucSocket(),
+		"JUJU_AGENT_SOCKET_ADDRESS=" + paths.GetJujucSocket().Address,
```

Comment:
Offhand, I don't think there's any upgrade issues in renaming these, but could we just think it through and ensure existing juju models which are upgraded will be ok.

Reply:
test IAAS, model upgrade from 2.6.6 to 2.7

Reply:
tested bootstrap to LXD on 2.6.7 then 

- deploy postgresql;
- upgrade controller/model;
- test `juju run` and `action`;